### ISSUE TITLE
Don't display completed-but-repeatable quests as "Completed"

### ISSUE THREAD

**Issue by SXRWahrheit** [FlyingPikachu/Quests#92]
(__Required__) What is your Quests version / build number (do _not_ say "latest")?
b169

(__Required__) Please describe the problem and how to replicate it. Any console errors?
We've had a previous discussion about this. Currently, a completed repeatable quest will result in no particle effects above NPCs and a listing as "Completed" on the NPC's quest display, with no indication that the quest is repeatable.

Desired behavior is a configuration option (in main plugin config or individual quest options) to display the quest as-if-new or otherwise indicate that it is repeatable both in chat options and with particle effects so that players do not inadvertently miss out on available repeatable quests.

(__Optional__) Can you provide an example quest from quests.yml _using Pastebin.com_?
```yaml
  Daily Fishing:
    name: Demonstrate your fishing skills (Daily)
    redo-delay: 43200
    npc-giver-id: 11
    requirements:
      quests:
      - Teach the newbie to fish
      fail-requirement-message: '<gray>Fishing Newbie<white>: You need to teach me
        how to fish first!'
    ask-message: '<gray>Fishing Newbie<white>: Could you show me a little more of
      your fishing skills?'
    finish-message: '<gray>Fishing Newbie<white>: Awesome! Maybe someday I''ll be
      as good as you are!'
    stages:
      ordered:
        '1':
          fish-to-catch: 16
    rewards:
      money: 50
      quest-points: 1
```

**Comment by FlyingPikachu** [comment]
This was brought up once before in the Blackvein repo (can't find the original). As I recall, I came to the conclusion that the "Completed" message is to indicate that the quest has been _completed at least once_. I do not recall if there were plans to change this.

However, I agree that some action should be taken to visually inform users that a quest is redo-able. Your config option is one idea, but I'd like leave this open for a while pending further ideas.

**Comment by SXRWahrheit** [comment]
As long as a repeatable-but-completed quest is visually distinct from a not-repeatable-and-completed quest, that's fine with me. I think the VFX should default to enabled for available repeatable quests, or there should at least be an option.

**Comment by OwnAgePau** [comment]
Maybe its an option to only show it visually once to indicate that the user has indicated it alteast once.

**Comment by FlyingPikachu** [comment]
I've given it some thought and agree that @SXRWahrheit 's proposal is best, which boils down to:

1. Add a config option to show particle effects for redoable quests
2. Display a sort of "(Redoable)" tag in, at least, the NPC GUI

How to execute these is a bit tricky. Obviously, not every admin is going to want a NPC effect for quests that are redoable (in other words, they're happy with the current setup). That said, we can't just disable the `show-npc-effects` option since that would disable effects for new quests, too. We could add a separate `show-redoable-npc-effects`, but then admins might enable that while disabling the base `show-npc-effects`, which may cause confusion for players across servers.

All that said, I think it's time to redesign config.yml. We're already saving the config in a manner which allows comments, so I'm thinking [something like this](https://pastebin.com/Ksjx59yp). Please check out line 36 and let me know what y'all think.

For the "(Redoable)" tag... it should be noted that the "(Completed)" message currently only shows for quests which are given by a NPC whom does not have a GUI display. It does **not** show in GUIs or when considering quests through _/quests list_ (although I guess showing that in the latter might defeat the purpose of _/quests stats_). So we ought to consider the implications of adding the tags elsewhere if it might make one of those commands rather useless. Thoughts?

**Comment by SXRWahrheit** [comment]
Line 36 etc. looks great.

I'm not familiar with the GUI display that you're talking about. Right now, players right click on an NPC and get a list of quests. Is there an alternative?

I put daily/weekly/etc in the names of all of the quests on my server, so I don't really need the addition of a (redoable) tag - if anything I would like the option to just exclude the (Completed) tag on redoable quests. Adding a redoable tag would be fine, and it's always possible to change the display functionality if someone comes up with a better idea, of course.

**Comment by FlyingPikachu** [comment]
@SXRWahrheit Yes, check out option 9 when creating/editing a quest :)

It sounds like the plan is to **replace** (Completed) with (Redoable) when applicable. Still gotta determine the fate of _/quests list_ vs _/quests stats_, but I'll add "help wanted" until I start work on this myself.

**Comment by SXRWahrheit** [comment]
The word "redoable" itself feels weird, by the way. Any suggested replacements would be welcomed.

**Comment by FlyingPikachu** [comment]
Agreed. That word was used internally before I started work on the plugin. How about "repeatable"?

**Comment by SXRWahrheit** [comment]
Repeatable is much better.

**Comment by FlyingPikachu** [comment]
Updated config per discussion. The new options don't do anything in code yet. I was wrong about being able to add comments to it, so I merely added a header to direct users to the Wiki.

**Comment by FlyingPikachu** [comment]
A new "repeatable" string has been added to the language file and will show at the quest accept prompt for NPCs without a GUI. To clarify, the "(Completed)" tag shown in the image below will now display "(Repeatable)" instead.

![Uploading 1.png…]()

I have not deleted the "completed" string from the lang file as it is still used the moment a user completes a quest in-game.

**Comment by SXRWahrheit** [comment]
That sounds great!

**Comment by FlyingPikachu** [comment]
@SXRWahrheit If a NPC has both repeatable AND uncompleted quests, which particle effect should show?

**Comment by SXRWahrheit** [comment]
Priority goes to uncompleted, IMO.

**Comment by SXRWahrheit** [comment]
I plan to use the same particle for both (maybe that'll change in the future), but I imagine "you haven't done this yet" is more important than "you can do this again."

**Comment by FlyingPikachu** [comment]
The repeat effect has been implemented and priority goes to uncompleted quests ("new-effect" in config). I changed the repeat-effect in config from "witch" to "angry_villager", made it so the capitalization of effect names no longer matters, and cut down on error spam if an invalid effect is used in config.

The only things left are to add the "(Repeatable)" tag to a few places where there is no tag at all, and to determine the fate of _/quests list_ vs _/quests stats.

**Comment by FlyingPikachu** [comment]
![1](https://user-images.githubusercontent.com/2267126/29695084-90f13eaa-890e-11e7-9690-c6276c306789.png)

In reference to changing this setup^, I've given it more thought and am now planing to revert replacing (Completed) with (Repeatable). As I mentioned in my first reply to this issue, the mere presence of a tag means it has been _completed at least once_, which I feel "Completed" makes clear. While the fact that the quest appears in the quest prompt _means_ it's repeatable, the word "repeatable" doesn't expressely imply to unfamiliar players that they've done the quest at least once already.

On the flip side, @SXRWahrheit has made a good point that having completed quests grouped in the same list as uncompleted quests isn't visually distinct, in which case "Repeatable" makes more sense. Since SXR also mentioned, "I would like the option to just exclude the (Completed) tag" my new plan is to simply keep the tag configurable in the language file, so that it can be changed from "(Completed)" to "(Repeatable)", or removed entirely, per admin's preference.

Sorry for the wall of text but I felt it best to explain why I'm changing "Repeatable" back to "Completed" now rather than later.

**Comment by FlyingPikachu** [comment]
Fixed a bug where repeatable quests would be listed multiple times in _/quests stats_. Also completed unfinished Blackvein code which tracks the number of times the quests have been taken and shows that with a trailing number. That said, I've decided to leave _stats_ and _list_ as separate commands for now, although they could certainly be combined one day.

With that, I believe all tasks for this issue have been completed. Thanks!
